src/gui/screen_menu_filament.cpp

```diff
@@ -30,36 +30,60 @@ const menu_item_t _menu_filament_items[] = {
     { { "Purge Filament", 0, WI_LABEL }, SCREEN_MENU_NO_SCREEN },
 };
 
-static void _load_ena(screen_t *screen) {
-    psmd->items[MI_LOAD].item.type &= ~WI_DISABLED;
-}
-static void _change_dis(screen_t *screen) {
-    psmd->items[MI_CHANGE].item.type |= WI_DISABLED;
+template <MI_t ENUM>
+static void disable_item(screen_t *screen) {
+    if (!(psmd->items[ENUM].item.type & WI_DISABLED)) {
+        psmd->items[ENUM].item.type |= WI_DISABLED;
+        psmd->menu.win.f_invalid = 1;
+    }
 }
-static void _change_ena(screen_t *screen) {
-    psmd->items[MI_CHANGE].item.type &= ~WI_DISABLED;
+template <MI_t ENUM>
+static void enable_item(screen_t *screen) {
+    if (psmd->items[ENUM].item.type & WI_DISABLED) {
+        psmd->items[ENUM].item.type &= ~WI_DISABLED;
+        psmd->menu.win.f_invalid = 1;
+    }
 }
 
+/*
```

Comment:
when you already have such a nice table, it would be better if it used the doxygen markings